### PR TITLE
1008 prevent negative numbers

### PR DESCRIPTION
**Pull request by hasham-qaiser**
re push/re-opened pull request.

### CODE REVIEW

**Comment by whitneywind** [comment]
- Hey, @hasham-qaiser this looks good overall! It effectively prevents a negative age from being entered and adds a helpful error message, too. I have a few questions for you.
- First, is the UseStyles fn in the Input showing up as an error on your end too? It doesn’t look like your code touched this part at all, so just wondering if it looks like that for you too. If it’s not just me, it could warrant a future issue to fix it.
- Also I’m not sure I understand why line 38 was deleted in IntroductionStep.tsx. Without it, the age defaults back to 0 instead of pulling a previously entered age from the FormStateContext (if one exists). The other inputs on the page pull their defaults from the state if they exist, so I’m not sure why this one wouldn't. If there is a reason I’m not understanding, please let me know!

**Comment by hasham-qaiser** [comment]
> * Also I’m not sure I understand why line 38 was deleted in IntroductionStep.tsx. Without it, the age defaults back to 0 instead of pulling a previously entered age from the FormStateContext (if one exists). The other inputs on the page pull their defaults from the state if they exist, so I’m not sure why this one wouldn't. If there is a reason I’m not understanding, please let me know

Didn't see that it was affecting that! i can add it back and make another push

**Comment by hasham-qaiser** [comment]
> * Hey, @hasham-qaiser this looks good overall! It effectively prevents a negative age from being entered and adds a helpful error message, too. I have a few questions for you.
> * First, is the UseStyles fn in the Input showing up as an error on your end too? It doesn’t look like your code touched this part at all, so just wondering if it looks like that for you too. If it’s not just me, it could warrant a future issue to fix it.
> * Also I’m not sure I understand why line 38 was deleted in IntroductionStep.tsx. Without it, the age defaults back to 0 instead of pulling a previously entered age from the FormStateContext (if one exists). The other inputs on the page pull their defaults from the state if they exist, so I’m not sure why this one wouldn't. If there is a reason I’m not understanding, please let me know!

added back line 48*. Hope everything is good for a merge now!

**Review by whitneywind — APPROVED** [review]

File: products/statement-generator/src/components/Input.tsx [hunk 1]
```diff
@@ -101,16 +102,39 @@ const InputArea: React.FC<InputFieldProps> = ({
...
   const utilityClasses = useUtilityStyles();
   const classes = useStyles({
     disabled,
     useShort: shortWidth || type === 'number',
   });
 
-  const [valid, isValid] = useState(false);
-  const checkValid = (e: string) => {
-    isValid(e.length > 0);
+  const [valid, isValid] = useState(true);
```
Comment (sydneywalcoff) [inline products/statement-generator/src/components/Input.tsx:113]: can you rename the setter here? It's a little confusing to have isValid when you already have a valid variable.

File: products/statement-generator/src/components/Input.tsx [hunk 2]
```diff
@@ -101,16 +102,39 @@ const InputArea: React.FC<InputFieldProps> = ({
...
+
+  const [value, setValue] = useState(defaultValue || '');
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const inputValue = e.currentTarget.value;
+
+    if (type === 'number') {
```
Comment (sydneywalcoff) [inline products/statement-generator/src/components/Input.tsx:120]: This is very creative, but I think it might be a little much. I think we might be able to use the inherent ['min' attribute](https://www.w3schools.com/tags/att_input_min.asp) on the input to prevent negative values. Since our input component is utilizing the OutlinedInput component from MUI, check out the [inputProps](https://mui.com/material-ui/api/outlined-input/#OutlinedInput-prop-inputProps) to apply the min value. 

bonus points if you want to hide the down arrow when value == 0, but we can always have that as a future issue.

**Review by sydneywalcoff — CHANGES_REQUESTED** [review]
Hi @hasham-qaiser ! Thanks for your PR. I added some comments.

File: products/statement-generator/src/components/Input.tsx (hunk 2 above)
Comment (hasham-qaiser) [reply products/statement-generator/src/components/Input.tsx]: Just made these changes, tried to go for the bonus points but i may have to open another issue for that as it might need some css styling.

**Review by sydneywalcoff — CHANGES_REQUESTED** [review]
@hasham-qaiser This looks good! Just one little thing. 

Can you change the min to 0, so it matches the placeholder? Otherwise when you click down it changes from 0 to 1.

[Check out this video](https://watch.screencastify.com/v/VqF0FtH0NQPjGRzCgZZO)

**Comment by hasham-qaiser** [comment]
> @hasham-qaiser This looks good! Just one little thing.
> 
> Can you change the min to 0, so it matches the placeholder? Otherwise when you click down it changes from 0 to 1.
> 
> [Check out this video](https://watch.screencastify.com/v/VqF0FtH0NQPjGRzCgZZO)

done!

File: products/statement-generator/src/components/Input.tsx [hunk 3]
```diff
@@ -88,6 +88,7 @@ interface InputFieldProps {
   adornment?: string;
   className?: string;
   shortWidth?: boolean; // if true, element will have a set width
+  min?: number;
```
Comment (sydneywalcoff) [inline products/statement-generator/src/components/Input.tsx:91]: here

File: products/statement-generator/src/components/Input.tsx [hunk 4]
```diff
@@ -101,6 +102,7 @@ const InputArea: React.FC<InputFieldProps> = ({
   adornment,
   className = '',
   shortWidth = false,
+  min,
```
Comment (sydneywalcoff) [inline products/statement-generator/src/components/Input.tsx:105]: here

File: products/statement-generator/src/pages-form/IntroductionStep.tsx [hunk 5]
```diff
@@ -37,6 +37,7 @@ export function IntroductionStep() {
           defaultValue={fullName}
           shortWidth
           handleChange={onInputChange}
+          min={0}
```
Comment (sydneywalcoff) [inline products/statement-generator/src/pages-form/IntroductionStep.tsx:40]: here

File: products/statement-generator/src/pages-form/IntroductionStep.tsx [hunk 6]
```diff
@@ -48,6 +49,7 @@ export function IntroductionStep() {
           defaultValue={age}
           placeholder="0"
           handleChange={onInputChange}
+          min={1}
```
Comment (sydneywalcoff) [inline products/statement-generator/src/pages-form/IntroductionStep.tsx:52]: here

**Review by sydneywalcoff — CHANGES_REQUESTED** [review]
@hasham-qaiser The logic looks good! it's performing as it should. 

I think we still have some code to clean up, though.

Screen Shot 2023-11-06 at 4 39 24 PM
I noticed that the min prop you pass into the Input component wasn't being used. I think that because the InputProps prop we're using, implements an inherent value on the input element, we don't need to define or type it.
Screen Shot 2023-11-06 at 4 40 54 PM


TLDR; I think line 136 on the Input component is all we need to accomplish this. I marked the others to be deleted.

**Review by sydneywalcoff — APPROVED** [review]
@hasham-qaiser yay! Thank you! Works great, looks great! I'll merge now!
